www/src/Services/Http.ts

```diff
@@ -0,0 +1,43 @@
+// This is modeled after Axios to make an easier transition, so this may need more buildout
+// if more than simple GET and POST requests are required.
+
+class Http {
+	async get(url: string, headers: object = {}) {
+		try {
+			const response = await fetch(url, {
+				method: 'GET',
+				headers: {
+					'Content-Type': 'application/json',
+					...headers,
+				},
+			});
+
+			const json = await response.json();
+			return Promise.resolve({ data: json });
+		}
+		catch (err) {
+			return Promise.reject(err);
+		}
+	}
+
+	async post(url: string, body: unknown, headers: object = {}) {
+		try {
+			const response = await fetch(url, {
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/json',
+					...headers,
+				},
+				body: JSON.stringify(body),
+			});
+
+			const json = await response.json();
+			return Promise.resolve({ data: json });
+		}
+		catch (err) {
+			return Promise.reject(err);
+		}
+	}
+}
```

Comment:
if it is bundle size you want to save, post and get are nearly identical you can expose them as the same function wrapped with an option parameter:

```ts
async post(url: string, body: unknown, headers: object = {}) {
  this.fetchWrapper(url, {
    headers,
    method: 'POST',
    body: JSON.stringify(body),
  });
}
```

where fetch wrapper is basically the body of post or get with some extra spread operator in the optios

Reply:
yes this would have give you at most other 200 bytes less. at most. between 100 and 200 probably. but felt like repeated code, so that was nice too

Reply:
A better argument for consolidating the `fetch` calls is that data transformation and error handling is the same for `get()` and `post()`. The next time I'm working on web updates I'll take a looking at optimizing this.